packages/devtools_app/lib/src/ui/utils.dart

```diff
@@ -227,3 +228,39 @@ class ColorPair {
 
   final Color background;
 }
+
+class ThemedColorPair {
+  const ThemedColorPair({@required this.background, @required this.foreground});
+
+  factory ThemedColorPair.from(ColorPair colorPair) {
+    return ThemedColorPair(
+      foreground: ThemedColor.fromSingle(colorPair.foreground),
+      background: ThemedColor.fromSingle(colorPair.background),
+    );
+  }
+
+  final ThemedColor foreground;
+
+  final ThemedColor background;
+}
+
+/// A theme-dependent color.
+///
+/// When possible, themed colors should be specified in an extension on
+/// [ColorScheme] using the [ColorScheme.isLight] getter. However, this class
+/// may be used when access to the [BuildContext] is not available at the time
+/// the color needs to be specified.
+class ThemedColor {
+  const ThemedColor({this.light, this.dark});
+
+  factory ThemedColor.fromSingle(Color color) =>
```

Comment:
this should be a const constructor as well which will enable a lot more of the ThemedColor to be const as well.

Reply:
factory constructors can't be const since Dart requires that const constructors have no body

Reply:
Ah nevermind - I can make this const and remove the factory